quantum/action_util.c

```diff
@@ -295,6 +298,53 @@ void send_keyboard_report(void) {
 #endif
 }
 
+#ifdef NKRO_ENABLE
+void send_nkro_report(void) {
+    nkro_report->mods = real_mods | weak_mods;
+
+#    ifndef NO_ACTION_ONESHOT
+    if (oneshot_mods) {
+#        if (defined(ONESHOT_TIMEOUT) && (ONESHOT_TIMEOUT > 0))
+        if (has_oneshot_mods_timed_out()) {
+            dprintf("Oneshot: timeout\n");
+            clear_oneshot_mods();
+        }
+#        endif
+        nkro_report->mods |= oneshot_mods;
+        if (has_anykey()) {
+            clear_oneshot_mods();
+        }
+    }
+#    endif
+
+#    ifdef KEY_OVERRIDE_ENABLE
+    // These need to be last to be able to properly control key overrides
+    nkro_report->mods &= ~suppressed_mods;
+    nkro_report->mods |= weak_override_mods;
+#    endif
```

Comment:
The duplication of this code between `send_6kro_report()` and `send_nkro_report()` does not look good.  Maybe introduce a helper function like `static uint8_t collect_mods_for_report(void)` and move this code there?